### DISCUSSION TITLE
How do I upload file in the browser env?

### DISCUSSION THREAD

**Discussion by Sayid1** [minio/minio#11388] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I needed some problems when Using JavaScript SDK. I uploaded the file in the browser environment and it could be uploaded normally, but after downloading, tell me the file is damaged. Please help me check the problem, thank you very much.

**Describe the solution you'd like**
Please help me how to upload the file in the browser, thank you

**Describe alternatives you've considered**
No yet

**Additional context**
Code are here
```
var reader = new FileReader()
reader.readAsArrayBuffer(file)
reader.onload = function () {
  minioClient
    .putObject('bucket', 'file.png', reader.result)
    .then(console.log)
    .catch(console.error)
}
```

**Comment by harshavardhana** [comment, marked as answer]
you can look at the following example https://github.com/harshavardhana/minio-js-browser-upload
